feat(ui): Implement bulk move for files and folders

ManojTauro opened the pull request:
### What changes are being made and why?

This PR implements the ability to move multiple files and folders simultaneously within the namespace file editor. This addresses the feature request in #5512 and provides a more efficient way for users to organize their files.

**Key changes include:**

   - Added a "Move" option to the context menu for selected files/folders in the file tree.
   - Implemented a new dialog for selecting the destination folder, which now includes an explicit "Root Folder" option.
   - Ensured sequential processing of move operations on the frontend to prevent potential conflicts or race conditions with the backend API.
   - Fixed a bug where the file selection was being cleared prematurely due to a global click listener, which prevented bulk operations from working correctly.
 
   closes #5512

Commit:
feat(ui): Implement bulk move for files and folders

elevatebart commented on ui/src/components/inputs/EditorSidebar.vue:
```diff
@@ -310,6 +313,47 @@
+                    v-for="folder in folders"
+                    :key="folder"
+                    :value="folder"
+                    :label="folder === '__ROOT__' ? $t('namespace files.dialog.root_folder') : folder"
```
What happens if one creates a folder called `__ROOT__`?
Did you check?

ManojTauro replied:
Yes, that might be confusing to users. How about if i represent root folder as "." ?

elevatebart replied:
Or any character not fit to be a file or folder name:

`#` or `$` are good

ManojTauro replied:
okay, I'll try # then. Thanks @elevatebart

elevatebart commented on ui/src/components/inputs/EditorSidebar.vue:
```diff
@@ -735,6 +797,75 @@
+                let successCount = 0;
+                let errorCount = 0;
+
+                for (const item of itemsToMove) {
+                    try {
+                        const newPath = this.moveDialog.folder === "__ROOT__"
+                            ? item.node.fileName
+                            : `${this.moveDialog.folder}/${item.node.fileName}`;
+
+                        await this.moveFileDirectory({
+                            namespace: this.namespaceId,
+                            old: item.path,
+                            new: newPath,
+                            type: item.node.type,
+                        });
+
+                        successCount++;
```
You do not need to wait for one file move to be over to process the next. 
Could you use a call to `Promise.all()` to batch all the file moves?

I believe this might greatly accelerate the file processing.

As a side note: Can you please create an issue to update the API? we need to send all file/folder move requests in a bulk request. Asking for each of theme independently seems inefficient.

ManojTauro replied:
Sure. I have replaced code with Promise.all() approach.

Also, I didn't get you on the creation of issue to update the API.

elevatebart commented on ui/src/components/inputs/EditorSidebar.vue:
```diff
@@ -517,8 +567,20 @@
                     this.selectedFiles = flatList.slice(start, end + 1).map(item => item.path);
                     this.selectedNodes = flatList.slice(start, end + 1).map(item => item.id);
+
+                } else if (window.event.ctrlKey || window.event.metaKey) {
```
This will return true on windows, if the windows key is entered. 
Is this intened?

If not, you can always ask for the userAgents OS and deal with it this way.

ManojTauro replied:
No, this is basically to check if user is pressing ctrl or cmd(mac) key.

elevatebart commented on ui/src/components/inputs/EditorSidebar.vue:
```diff
@@ -1163,10 +1294,8 @@
             },
         },
         mounted() {
-            document.addEventListener("click", this.clearSelection);
         },
         beforeUnmount() {
-            document.removeEventListener("click", this.clearSelection);
         },
```
Why was this removed? WAs it replaced by anything? How do you deselect files when clicking outside?

ManojTauro replied:
This was removed because, when I selected the files/folders and then perform right click to select Move files option. After selecting Move Files button the files were detected automatically. But i see that it's affecting the deselect functionality when we click outside of the files.
I'll see how can i handle this and bring these listeners back.

ManojTauro replied:
Okay, Now I have restored listeners. And added one more variable to track selectedFiles and cleared this after performing move operation.

elevatebart reviewed (changes requested):
Thank you much @ManojTauro 

This feature will be really useful so the PR is very much welcome.
Before we merge I have a few questions and comments if you have a minute.

Cheers

Commit:
Changes:
- '#' to be considered as a root folder.
- Move files using Promise.all to improve performance

MilosPaunovic committed:
Merge branch 'develop' into feat/issue-5512

MilosPaunovic commented:
Hello @ManojTauro, could you please resolve the merge conflict before we can proceed with the review?

MilosPaunovic commented:
Hey @ManojTauro, are there any updates on this?

ManojTauro commented:
> Hey @ManojTauro, are there any updates on this?

Hey @MilosPaunovic ,
Really sorry about the delay. I'll resolve the conflicts this weekend.
Again, apologies, I missed the comments.

ManojTauro committed:
Merge branch 'develop' into feat/issue-5512

elevatebart committed:
Merge branch 'develop' into feat/issue-5512

MilosPaunovic commented:
Hey @ManojTauro, there are again some conflicts, could you handle those, as well, please? 🙏🏻

MilosPaunovic commented:
Closing, due to inactivity. 

Thanks for your efforts on it @ManojTauro! 🍀
